src/parseDat.js

```diff
@@ -14,51 +18,110 @@ function parseLine(line, fields, knex, st) {
+          console.log(values[name], value);
+          throw new Error(
+            `Failed to parse value for field ${name}. Line:\n${line}`,
+          );
         }
       } else if (type === "integer") {
         values[name] = parseInt(value, 10);
         if (Number.isNaN(values[name])) {
-          throw new Error(`Failed to parse value for field ${name}. Line:\n${line}`);
+          throw new Error(
+            `Failed to parse value for field ${name}. Line:\n${line}`,
+          );
         }
       } else if (type === "date") {
         if (value.length !== 8) {
-          throw new Error(`Invalid value ${value} for field ${name}. Line:\n${line}`);
+          throw new Error(
+            `Invalid value ${value} for field ${name}. Line:\n${line}`,
+          );
         }
-        values[name] = `${value.slice(0, 4)}-${value.slice(4, 6)}-${value.slice(6, 8)}`;
+        values[name] = `${value.slice(0, 4)}-${value.slice(4, 6)}-${value.slice(
+          6,
+          8,
+        )}`;
       } else {
         values[name] = value;
       }
     }
     index += length;
   });
-  if (values.lat && values.lon) {
-    values.point = st.geomFromText(`Point(${values.lon} ${values.lat})`, 4326);
-  }
-  if (values.x && values.y) {
-    values.point = knex.raw(`ST_Transform(ST_GeomFromText('Point(${values.x} ${values.y})',2392),4326)`);
+
+  if (typeof values.lat !== "undefined" && typeof values.lon !== "undefined") {
+    values.point = st.geomFromText(
+      `POINT(${_.get(values, "lon", 0)} ${_.get(values, "lat", 0)})`,
+      4326,
+    );
+  } else if (
+    typeof values.x !== "undefined" &&
+    typeof values.y !== "undefined"
+  ) {
+    values.point = knex.raw(
+      `ST_Transform(ST_GeomFromText('POINT(${_.get(values, "x", 0)} ${_.get(
+        values,
+        "y",
+        0,
+      )})',2392),4326)`,
+    );
   }
+
   return values;
 }
 
-function parseDat(filename, fields, knex, tableName, trx, st) {
+function getIndexForTable(tableName) {
+  const tableSchema = _.get(schema, tableName, false);
+  const indices = _.get(tableSchema, "fields", []).reduceRight(
+    (indexNames, field) => {
+      const name = _.get(field, "name", "");
+
+      // If this field is an unique index, we're interested in it. Do not add
+      // non-unique indices here.
+      if (
+        (_.get(field, "primary", false) || _.get(field, "unique", false)) &&
+        name
+      ) {
+        indexNames.push(name);
+      }
+
+      return indexNames;
+    },
+    [],
+  );
+
+  return indices;
+}
+
+function parseDat(filename, fields, knex, tableName, st) {
+  const indexColumns = getIndexForTable(tableName);
+
   const insertLines = async (lines) => {
-    console.log(`Inserting ${lines.length} lines from ${filename} to ${tableName}`);
-    await knex.withSchema("jore").transacting(trx).insert(lines).into(tableName);
+    console.log(
+      `Inserting ${lines.length} lines from ${filename} to ${tableName}`,
+    );
+
+    await upsert({
```

Comment:
Upsert = insert or update. Inserts the record if one doesn't exist, or updates it with new data if does. Uses the indices of the table to check for conflicting records.